amqp-postgres/amqp_postgres/amqp_consumer.py

```diff
@@ -0,0 +1,62 @@
+########
+# Copyright (c) 2018 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+############
+
+import json
+from uuid import uuid4
+
+from flask import current_app
+
+
+class AMQPLogsEventsConsumer(object):
+    LOGS_EXCHANGE = 'cloudify-logs'
+    EVENTS_EXCHANGE = 'cloudify-events'
+
+    def __init__(self, message_processor):
+        self.queue = str(uuid4())
```

Comment:
the queue name should likely be non-random
the idea is that if this program crashes, messages will still be routed to the queue, and after the program starts again, it will be able to consume them
for this, it must use the same queue name, and the queue must be durable, no auto delete, not exclusive (so the queue is set correctly, just the name is random)